app/src/main/java/com/example/drawingApp/DialogUtility.kt

```diff
@@ -0,0 +1,157 @@
+package com.example.drawingApp
+
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.core.view.isVisible
+import androidx.core.widget.addTextChangedListener
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.bottomsheet.BottomSheetDialog
+
+object DialogUtility {
+
+    const val MAX_CHAR_LEN = 16
+
+    /**
+     * used to updated the sheet adapter with the onEdit lambda
+     * the sheet is for dismissing to avoid window leaks
+     */
+    data class SheetObject(val onEdit: (Int, String) -> Unit, val sheet: BottomSheetDialog)
```

Comment:
Nit: Lambdas should be the last argument so you can use the special syntax with the {} on the outside